test/src/folding/parsers/highlight_parser_test.dart

```diff
@@ -10,142 +10,126 @@ import 'package:flutter_code_editor/src/single_line_comments/single_line_comment
 import 'package:flutter_test/flutter_test.dart';
 import 'package:highlight/highlight_core.dart';
 import 'package:highlight/languages/java.dart';
+import 'package:highlight/languages/scala.dart';
 
 void main() {
-  group('HighlightFoldableBlockParser', () {
-    test('parses', () {
-      final examples = [
-        //
-        _Example(
-          'Java. Empty string',
-          code: '',
-          mode: java,
-          expected: const [],
-          expectedInvalid: const [],
-        ),
-
-        _Example(
-          'Java. Multiline pair-character blocks',
-          code: '''
-public class MyClass {
-  public static void main(
-    String[
-    ] args) {
-    }
+  _testJava();
```

Comment:
This file is getting large. I suggest you split it by languages, make independent files with their `main`s, and put `parseAndCheck`, `FB`, and `T` to a shared file.